polkadot/node/core/pvf/src/security.rs

```diff
@@ -348,6 +361,50 @@ async fn check_seccomp(
 	}
 }
 
+/// Check if we can call `clone` with all sandboxing flags, and return an error if not.
+///
+/// We do this check by spawning a new process and trying to sandbox it. To get as close as possible
+/// to running the check in a worker, we try it... in a worker. The expected return status is 0 on
+/// success and -1 on failure.
+async fn check_can_do_secure_clone(
+	#[cfg_attr(not(target_os = "linux"), allow(unused_variables))]
+	prepare_worker_program_path: &Path,
+) -> SecureModeResult {
+	cfg_if::cfg_if! {
+		if #[cfg(target_os = "linux")] {
+			match tokio::process::Command::new(prepare_worker_program_path)
```

Comment:
AFAIR we're running workers in some kind of clean environment (pruning environment variables, etc.) In theory, running this check in a different environment could result in a different outcome of the check (think about `LD_PRELOAD`, `LD_LIBRARY_PATH`, and other things that cannot be talked about in polite society). I can hardly imagine how it can be an attack vector, but it still could be considered to run the check and the worker itself in the same environment; it's just safer.

Reply:
Hmm that complicates things as we don't know here what other security features are available and will be enacted. It seems reasonable to always clear the env vars in all the checks, and also have that be the first sandboxing step done - anything else we can do?

Reply:
I believe that's enough, and it's not that critical anyway, can be a follow-up.

Reply:
Makes sense to do it here. We will miss the next merge window, which is in less than three hours - which means we can take our time with this PR. :)